config/scripts/check.sh

```
@@ -365,43 +480,104 @@ check_cpu_limits() {
-        echo "[Phase 4/4] Checking stress-ng-cpu processes..."
-        
+        if [ "${guest_os}" = "windows" ]; then
+            echo "[Phase 4/4] Bootstrapping and checking Windows CPU burn workers (CNV_CPU_BURN=1)..."
+
+            # Build the bootstrap script and encode it as base64 UTF-16LE
+            # to use powershell -EncodedCommand, bypassing all SSH quoting issues.
+            local ps_script
+            # shellcheck disable=SC2016
+            ps_script='$n = (Get-CimInstance Win32_ComputerSystem).NumberOfLogicalProcessors; '
+            ps_script+='$workerCmd = "powershell.exe -NoProfile -Command `$env:CNV_CPU_BURN=1; [double]`$x=1; while(`$true){`$x=[math]::Sqrt(`$x+1)}"; '
+            ps_script+='1..$n | ForEach-Object { $r = Invoke-CimMethod -ClassName Win32_Process -MethodName Create -Arguments @{CommandLine=$workerCmd}; if($r.ReturnValue -ne 0){"WORKER_FAIL:rc=$($r.ReturnValue)"} }; '
```

Shellcheck (0.11.0)

[info] Expressions don't expand in single quotes, use double quotes for that.

(SC2016)